opengrok-indexer/src/main/java/org/opengrok/indexer/framework/PluginFramework.java

```diff
@@ -216,7 +216,8 @@ private PluginType loadClass(String classname) throws ClassNotFoundException,
 
         // check for implemented interfaces or extended superclasses
         for (Class intf1 : getSuperclassesAndInterfaces(c)) {
-            if (intf1.getCanonicalName().equals(classType.getCanonicalName())
+            if (intf1.getCanonicalName() != null && // anonymous classes
+                    intf1.getCanonicalName().equals(classType.getCanonicalName())
```

Comment:
See line 226 below which logs re `IAuthorizationPlugin` that is no longer applicable for all `PluginFramework` subclasses